independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/ComponentsProviderGenerator.java

```diff
@@ -387,16 +387,26 @@ void addComponentInternal(ObserverInfo observer) {
         private final ResultHandle removedBeansHandle;
         private final ClassOutput classOutput;
         private ResultHandle tccl;
+        // Shared annotation literals for an individual addRemovedBeansX() method
+        private final Map<AnnotationInstanceKey, ResultHandle> sharedQualifers;
```

Comment:
Is that worth it?

Reply:
I think it is. For example, because of SR Config almost every app will have ~ 30 removed beans with the `@ConfigProperty` qualifier, which in turn results in a new `AnotationLiteral` instance stored in the `removedBeans` list. This way, we can reduce the number of annotation literal instances to 2 (for the `hibernate-orm-quickstart`).